smart-contract-usage-in-javascript-sdk.md

```diff
@@ -0,0 +1,326 @@
+  public function init(available_count' : int, min_price' : int) : state =
+    { available_count = available_count', min_price = min_price', owner = Call.caller }
+
+
+  // function for anyone to reserve some count requiring available count and the minimum price payed
+  public stateful function reserve(count: int) =
+  
+    // requiring the price payed per item to be above or equal the set minimum price
+    require(Call.value >= state.min_price * count, "Payed price is less than minimum required")
+    let remaining_count = state.available_count - count
+    
+    // requiring count to be available
+    require(remaining_count >= 0, "Less available then to be reserved")
+    
+    // applying remaining count to the state
+    put(state{ available_count = remaining_count })
+
+
+  // function to get the remaining count, can be statically called as it doesn't change state
+  public function available_count() : int =
+    state.available_count
+
+
+  // function only to be called by the owner to withdraw all balance that was sent to the contract
+  public stateful function owner_withdraw() : int=
+  
+    // requiring contract caller to be the registered owner
+    require(Call.caller == state.owner, "Only owner can withdraw")
+    let withdrawed_balance = Contract.balance
+    
+    // creating spend of the balance to the registered owner
+    Chain.spend(state.owner, Contract.balance)
+    
+    // returning the balance that was withdrawed
+    withdrawed_balance
+```
+
+## SDK Contracts Workflow
+
+
+### 1. Initialize SDK client
+
+Inside the main function, to create the SDK client referencing the æternity network node and the needed public and private keys, add after the previously added lines:
+
+```
+// initialize aeternity sdk client
+const client = await Universal({
+    url: 'https://sdk-testnet.aepps.com', //replace with https://sdk-mainnet.aepps.com for mainnet
+    internalUrl: 'https://sdk-testnet.aepps.com', //replace with https://sdk-mainnet.aepps.com for mainnet
+    keypair: {
+        publicKey: 'REPLACE_WITH_PUBLIC_KEY',
+        secretKey: 'REPLACE_WITH_PRIVATE_KEY'
+    },
+    networkId: 'ae_uat', //replace with ae_mainnet for mainnet
+    nativeMode: true
+});
+```
+
+### 2. Reading the ExampleContract
+
+To read the contract code, add after the previously added lines:
+
+```
+// read the ExampleContract.aes from same directory
+const contract = fs.readFileSync('./ExampleContract.aes', 'utf-8');
+```
+
+### 3. Compiling the contract code
+
+To compile the contract code, add after the previously added lines:
+
+```
+// use the client to compile the contract code, catching eventual errors
+const compiled = await client.contractCompile(contract).catch(logError);
+
+// logging the bytecode output to the console to see the code work
+console.log('contract bytecode:', compiled.bytecode);
+```
+
+### 4. Deploying the contract bytecode to the blockchain
+
+To deploy the contract bytecode, add after the previously added lines:
```

Comment:
"Having the compiled contract bytecode, we are ready to deploy on the network of our choice: "